tests/yast2_gui/yast2_lan_typo.pm

```diff
@@ -0,0 +1,43 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that yast2-lan dose not crash if there are errors (like typos) in the ifcfg files.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base 'y2_module_guitest';
+use strict;
+use warnings;
+use testapi;
+use y2lan_restart_common qw(open_network_settings wait_for_xterm_to_be_visible close_xterm close_network_settings);
+use x11utils 'start_root_shell_in_xterm';
+use scheduler 'get_test_suite_data';
+
+sub check_typo {
+    my $error_in_ifcfg = shift;
+    assert_script_run($error_in_ifcfg);    # Insert an error in ifcfg file
+    open_network_settings;
+    close_network_settings;
+    wait_for_xterm_to_be_visible();
+}
+
+sub run {
+    my $test_data  = get_test_suite_data();
+    my $ifcfg_file = ' /etc/sysconfig/network/ifcfg-' . $test_data->{net_device};
+    record_info('IFCFG', 'Verify that putting wrong settings in ifcfg files do not provoke a crash');
+    start_root_shell_in_xterm();
+    assert_script_run("cat $ifcfg_file > backup");
+    foreach my $error_in_ifcfg (@{$test_data->{errors_in_ifcfg_file}}) {
+        check_typo($error_in_ifcfg . $ifcfg_file);    # See descriptions of errors in test_data
```

Comment:
same for the name of the method, I think that a duplicated entry is not a typo for example.
Could you pass the file as parameter so we can avoid that extra space pre-pending $ifcfg_file?

Reply:
I don't understand what you mean by "pass the file as parameter so we can avoid that extra space" ?

Reply:
you have an space prepending $ifcfg_file, which then is displayed as double when you do the 'cat' in line 34, if you pass the file you can do that concatenation inside the function.

Reply:
I still don't understand :(. I know there is a space prepending, without it we would have no space between the commands and the file name eg "sed -i 's/dhcp/dgcp/'/etc/sysconfig/network/ifcfg-eth0". Please explain what you mean by "pass the file".

Reply:
I think I figured out what you meant, I will push the change.

Reply:
Perfect, now is more clear that the methods is depending on the file to be passed so any command to use in test data will act somehow (regardless of its syntax) over that file. Could you please add named parameter?